lib/src/components/data-table/drag-and-drop/DragAndDropContainer.tsx

```
@@ -0,0 +1,81 @@
+import { useDataTable } from '../DataTableContext'
+import type { TableData } from '../DataTable.types'
+import {
+  closestCenter,
+  DndContext,
+  KeyboardSensor,
+  PointerSensor,
+  useSensor,
+  useSensors,
+  UniqueIdentifier
+} from '@dnd-kit/core'
+import { restrictToVerticalAxis } from '@dnd-kit/modifiers'
+import {
+  arrayMove,
+  sortableKeyboardCoordinates,
+  SortableContext,
+  verticalListSortingStrategy
+} from '@dnd-kit/sortable'
+import * as React from 'react'
+
+type DragAndDropContainerProps = {
+  idColumn?: string
+  onChange?: (oldIndex: number, newIndex: number, newData: TableData) => void
+}
+export const DragAndDropContainer: React.FC<DragAndDropContainerProps> = ({
+  idColumn = 'id',
+  onChange = undefined,
+  children
+}) => {
+  const { data, setData, setIsDragAndDrop } = useDataTable()
+
+  const rowOrder = React.useMemo(
+    () =>
+      data.results.map((row) => {
+        const id = row[idColumn]
+        if (id === undefined)
+          console.error(
```

Is it worth `throw`ing here? I assume this wouldn't allow drag and drop, and it would silently fail? I'm on the fence... but if we have a good Error Boundary wrapping each table then it should limit the impact 🤔

---

Weirdly, if some-but-not-all rows are missing IDs then it can still work. But that's a risky thing to rely on and I imagine it's likely that every row will be given its ID the same way and therefore if any are missing one, they all will be. Will throw and add a recommendation to use an `ErrorBoundary` in the docs.